x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_field/index.tsx

```diff
@@ -119,6 +126,28 @@ export const RuleActionsField: React.FC<Props> = ({ field, messageVariables }) =
     [field]
   );
 
+  const setActionFrequencyProperty = useCallback(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (key: string, value: any, index: number) => {
+      setTimeout(
+        () =>
+          field.setValue((prevValue: RuleAction[]) => {
+            const updatedActions = [...prevValue];
+            updatedActions[index] = {
+              ...updatedActions[index],
+              frequency: {
+                ...(updatedActions[index].frequency ?? DEFAULT_ACTION_FREQUENCY),
+                [key]: value,
+              },
+            };
+            return updatedActions;
+          }),
+        0
+      );
```

Comment:
Is it possible to not use `setTimeout(0)` here?

Reply:
The code block resembles the [block above](https://github.com/elastic/kibana/blob/756edc6328a615184c2be84b63213a83663e83e3/x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_field/index.tsx#L110). It'd be nice to get rid of `setTimeout` in both code blocks but it goes out of scope of this PR as for me.